libatalk/adouble/ad_attr.c

```diff
@@ -126,14 +171,32 @@ uint32_t ad_getid (struct adouble *adp, const dev_t st_dev, const ino_t st_ino ,
 
     if (adp) {
         if (sizeof(dev_t) == ad_getentrylen(adp, ADEID_PRIVDEV)) {
-            memcpy(&dev, ad_entry(adp, ADEID_PRIVDEV), sizeof(dev_t));
-            memcpy(&ino, ad_entry(adp, ADEID_PRIVINO), sizeof(ino_t));
-            memcpy(&a_did, ad_entry(adp, ADEID_DID), sizeof(cnid_t));
+            char *ade = NULL;
+            ade = ad_entry(adp, ADEID_PRIVDEV);
+            if (ade == NULL) {
+		return 0;
```

Comment:
type is uint32_t. Only caller for this function is `get_id()` in `etc/afpd/file.c`. Returning 0 here means that we skip the ad_setid() in the caller.

Reply:
I'm not terribly familiar with Netatalk internals. This mostly reflects what I could digest over about a 12ish hour period. It seems that returning 0 is probably the best I can do in terms of error handling here though because it will result in us not doing anything with the result.

That said, we should probably log this.

Reply:
Okay. I've looked this over some more, we need to skip lookup for ADEID_DID when AD_VERSION_EA. Will update PR accordingly later today (and add logging).